ios/FlankerNativeComponents/GameManager.swift

```diff
@@ -190,16 +193,38 @@ class GameManager {
   }
 
   func checkedAnswer(button: SelectedButton) {
-    invalidateTimers()
+    // Prevent and block multiple responses per trial
+    guard !hasResponded else {
+      print("[GameManager.swift] User has already responded in this trial.") // <---- if is a better tracer handling let me know.
+      return
+    }
+    hasResponded = true  // Mark that the user has responded
 
-    delegate?.setEnableButton(isEnable: false)
+    invalidateTimers()
 
     guard let gameParameters = gameParameters else { return }
-    guard
```

Comment:
explanation Why I Removed It:
The guard statement caused the checkedAnswe method to return early if any of the variables were nil.
This led to situations where clicking the button didn't register the event because respondTouchButton or endTrialTimestamp hadn't been set yet

Reply:
I think this is close to the cause but you're approach is backwards.

I understand the early return was problematic, but you should investigate the reason for nil values instead of arbitrarily adding values in here. The timing should be coming from elsewhere.